src/ucs/datastruct/list.h

```diff
@@ -129,6 +132,8 @@ ucs_list_is_first(const ucs_list_link_t *head, const ucs_list_link_t *elem)
 static inline int
 ucs_list_is_last(const ucs_list_link_t *head, const ucs_list_link_t *elem)
 {
+    UCS_ASAN_ADDRESS_IS_VALID(head, sizeof(ucs_list_link_t));
```

Comment:
maybe add helper macro UCS_LIST_CHECK_ELEM_IS_VALID
to avoid passing `sizeof(ucs_list_link_t)` every time